lib/puppet/module/task.rb

```diff
@@ -157,7 +185,9 @@ def implementations
     end
 
     def files
-      implementations.map {|imp| { 'name' => imp['name'], 'path' => imp['path'] } }
+      required_files = metadata.nil? ? [] : self.class.find_files(@name, metadata)
```

Comment:
This misses files included in each implementation. See the example at https://github.com/puppetlabs/puppet-specifications/tree/master/tasks#example-task-metadata.